### ISSUE TITLE
More whiteboard tools (SVG)

### ISSUE THREAD

**Issue by MzHub** [josephernest/bigpicture.js#10]
The ability to draw freehand and preset shapes (SVG) would be good for whiteboard-like use cases.

Hotkeys for different tools. Also panning could then happen with middle or right click, or something like space + drag from photoshop.

**Comment by Lambo87** [comment]
How do you save?

**Comment by josephernest** [comment]
@Lambo87 it depends: if you just want a quick save feature, just serialize the whole data with `localStorage`, this can be done in less than 10 lines, I did it in the past but I removed it, because I wanted this repository to focus on the rendering only.

If you want a more elaborated 'save' feature, you need user management, login/logout, etc. as I did for http://www.bigpicture.bi

What kind of 'save' are you interested in?

**Comment by Lambo87** [comment]
I just wanted to save and view the saved content for personal use

**Comment by josephernest** [comment]
Yes, but the problem is that it's impossible to easily save data on
filesystem with Javascript in a browser, for historical and security
reasons. It's very annoying, but it's a well known thing in web.

So the only solutions if you want to have a "save" option are :
* setting up a webserver+database like on http://bigpicture.bi
* or add a DOWNLOAD button that will serialize the data and allow to
download it as .html for example and save it to disk ; and also add a LOAD
buton that will load a .html file
